thirdparty/github_models.go

```diff
@@ -1,9 +1,58 @@
 package thirdparty
 
 import (
+	"encoding/json"
 	"time"
+
+	"github.com/pkg/errors"
 )
 
+const githubTimeFormat = "2006-01-02T15:04:05Z"
+
+type GithubTime time.Time
+
+// UnmarshalJSON is a helper to fix parsing of Github's Timestamps
+// Github time stamps look like this: 2018-02-07T19:54:26Z
+// strftime format: %Y-%m-%dT%H:%M:%SZ
+// Z indicates Zulu time (UTC)
+func (t *GithubTime) UnmarshalJSON(b []byte) error {
+	var timeStr string
+	if err := json.Unmarshal(b, &timeStr); err != nil {
+		return errors.Wrap(err, "expected JSON string while parsing time")
+	}
+	if len(timeStr) == 0 {
+		return errors.New("expected JSON string while parsing time")
+	}
+
+	loc, err := time.LoadLocation("UTC")
+	if err != nil {
+		return errors.Wrap(err, "error loading UTC time location")
+	}
+
+	newTime, err := time.ParseInLocation(githubTimeFormat, timeStr, loc)
+	if err != nil {
+		return errors.Wrapf(err, "Error parsing time '%s' in UTC time zone", timeStr)
+	}
+
+	*t = GithubTime(newTime)
+
+	return nil
+}
+
+func (t *GithubTime) MarshalJSON() ([]byte, error) {
+	timeStr := t.Time().Format(githubTimeFormat)
+
+	b := []byte{0x22}
+	b = append(b, []byte(timeStr)...)
+	b = append(b, 0x22)
```

Comment:
we've been talking offline about solutions, I think my current favorite is: 

    []byte(fmt.Sprintf(`"%s"`, timeStr))

but something that avoids raw hardcoded bytes

Reply:
done